### DISCUSSION TITLE
How Can I sort the column names , I am using Postgres and Deaver Enterprise 7.3.0.

### DISCUSSION THREAD

**Discussion by jota2406** [dbeaver/dbeaver#10670] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version
- DBeaver version

#### Connection specification:
- Database name and version

**Comment by uslss** [comment]
Hi @jota2406 ,
if I understand you correct, column order could be changed by clicking at Name column title in Table Metadata Editor
![2020-12-11_18h21_13](https://user-images.githubusercontent.com/31996417/101921234-d1498680-3bdd-11eb-84fd-064d913fc4a1.gif)

**Reply by detkliff** [reply]
> Hi @jota2406 , if I understand you correct, column order could be changed by clicking at Name column title in Table Metadata Editor ![2020-12-11_18h21_13](https://user-images.githubusercontent.com/31996417/101921234-d1498680-3bdd-11eb-84fd-064d913fc4a1.gif) [ ![2020-12-11_18h21_13](https://user-images.githubusercontent.com/31996417/101921234-d1498680-3bdd-11eb-84fd-064d913fc4a1.gif) ](https://user-images.githubusercontent.com/31996417/101921234-d1498680-3bdd-11eb-84fd-064d913fc4a1.gif) [ ](https://user-images.githubusercontent.com/31996417/101921234-d1498680-3bdd-11eb-84fd-064d913fc4a1.gif)

hello. Tell me please where is column ordering setting by default?

my ddl
CREATE TABLE data_mart_import.data_main (
	id int8 GENERATED BY DEFAULT AS IDENTITY NOT NULL,
	unom int4 NULL,
	file_address text NULL,
	nsi_address text NULL,
	nsi_address_simple text NULL,
	uik_number int4 NULL,
	district_id int4 NULL,
	io_number int4 NULL,
	"location" text NULL,
	address_addition text NULL,
	street_id int4 NULL,
	street_type_id int4 NULL,
	building_type text NULL,
	building_subtype text NULL,
	geodata json NULL,
	geodata_center json NULL,
	election_id int4 NULL,
	all_voters int4 NULL,
	is_voted int4 NULL,
	is_usual_voted int4 NULL,
	is_el_voted int4 NULL
)

But in interface i see

<img width="631" height="447" alt="image" src="https://github.com/user-attachments/assets/16a6e933-4233-4606-8475-b2f2be2ecdb6" />

without set up ordering

**Comment by jota2406** [comment]
I'm sorry, it's not what I was looking for, if I have a Table with three column : #1 Data1, #2 Data2, #3 Data3 I would like change to #1 Data3, #2 Data2 , #3 Data1. Thank you

**Reply by TB-HunBoka** [reply]
I might misunderstand what do you want to achieve, but what if you simple write the select query in that order?
Select Data3, Data2, Data1 from table.

**Comment by lemon-juice** [comment, marked as answer]
You can't change column order in postgres because it is not supported by the database. You can reorder columns in DBeaver when you view table data and save it as a filter to be remembered but this will only affect your DBeaver view, it will not be saved in the database.

**Reply by Theanderblast** [reply]
What if I don't want to change the underlying column order, but I just want to see the columns in a different/alphabetic order in the data view?   I usually pin columns that I'm interested, but for a table with many columns, it can be sometimes be hard to find a particular column.

Note, I don't need this to be supported by the database.  I don't want the database column order to change, just for DBeaver to change what I'm looking at.

**Comment by jota2406** [comment]
I using Sql Manager Lite for Postgres SQL 6.1.2 and I can reorder columns easily  in the database and I don't have  any problem to reorder columns , so I expected  to do the same in DBeaver.

**Comment by lemon-juice** [comment]
Yes, but it doesn't change the fact that Postgres does not support reordering columns. Sql Manager uses certain tricks to do it and as it is [stated on their web site](https://www.sqlmanager.net/en/products/postgresql/manager/documentation/hs5433) it does not work in all cases. At best you can request this feature from DBeaver devs but this may not be easy to implement unless done only for the most simple cases like tables without relations, etc.

**Comment by jota2406** [comment]
Yes, you are right, I going to request that because this feature it´s very important even only for  only tables without relations.
Thank you

**Comment by vmandrilly2** [comment]
Just let DBeaver save the ordering WE WANTED in some settings file somewhere on the computer. We can't just have to manually reorder every time we restart the software...

**Comment by emrecpp** [comment]
For a GUI-based solution to physically reorder columns in PostgreSQL, you can use EMS SQL Manager for PostgreSQL; by right-clicking in the Columns tab of the Table Editor and selecting Reorder Columns, you can visually rearrange your fields without writing manual SQL . It is important to understand that because PostgreSQL lacks a native command for this operation, the tool handles the process by performing a full 'table recreation' in the background, which involves creating a temporary table, migrating all existing data, and replacing the original table . While this automates the complex steps for you, the physical rewrite of every row can take a significant amount of time on large datasets and will lock the table during the process, so ensure you have a full backup and consider the potential downtime before applying these changes to production environments .
